subsys/net/ip/l2/ethernet/ethernet.c

```diff
@@ -757,18 +759,74 @@ int net_eth_vlan_disable(struct net_if *iface, u16_t tag)
 NET_L2_INIT(ETHERNET_L2, ethernet_recv, ethernet_send, ethernet_reserve,
 	    ethernet_enable);
 
+#if defined(CONFIG_NET_L2_ETHERNET_MGMT)
```

Comment:
> You don't need that actually. Now that you added a patch "stubbing" ethernet_mgmt_raise_carrier_on/off_event() it will build with or w/o net_mgmt.

Actually I did not had these checks here because of that. My reasoning was that net_eth_carrier_on() would be no-op if ethernet mgmt is turned off, but I suppose we can have that function to work regardless of mgmt support.

Reply:
yep, net_eth_carrier_on/off are not mgmt dependent. If mgmt is enabled it will raise the mgmt even (for whoever is listening to these), but that's all.